pkgs/applications/window-managers/sway/contrib.nix

```diff
@@ -0,0 +1,98 @@
+{ stdenv
+
+, fetchurl
+, coreutils
+, makeWrapper
+, sway-unwrapped
+, wl-clipboard
+, libnotify
+, slurp
+, grim
+, jq
+
+, python3Packages
+}:
+
+{
+
+grimshot = stdenv.mkDerivation rec {
+  pname = "grimshot";
+  version = "2020-05-08";
+  rev = "b1d08db5f5112ab562f89564825e3e791b0682c4";
+
+  # master has new fixes and features, and a man page
+  # after sway-1.5 these may be switched to sway-unwrapped.src
+  bsrc = fetchurl {
+    url = "https://raw.githubusercontent.com/swaywm/sway/${rev}/contrib/grimshot";
+    sha256 = "1awzmzkib8a7q5s78xyh8za03lplqfpbasqp3lidqqmjqs882jq9";
+  };
+
+  msrc = fetchurl {
+    url = "https://raw.githubusercontent.com/swaywm/sway/${rev}/contrib/grimshot.1";
+    sha256 = "191xxjfhf61gkxl3b0f694h0nrwd7vfnyp5afk8snhhr6q7ia4jz";
+  };
+
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  phases = [ "installPhase" "fixupPhase" "checkPhase" ];
+
+  installPhase = ''
+    install -D ${msrc} $out/share/man/man1/grimshot.1
+
+    install -Dm 0755 ${bsrc} $out/bin/grimshot
+    wrapProgram $out/bin/grimshot --set PATH \
+      "${stdenv.lib.makeBinPath [
+        sway-unwrapped
+        wl-clipboard
+        coreutils
+        libnotify
+        slurp
+        grim
+        jq
+        ] }"
+  '';
+
+  doCheck = true;
+
+  checkPhase = ''
+    # check always returns 0?
+    if [[ $($out/bin/grimshot check | grep "NOT FOUND") ]]; then false; fi
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A helper for screenshots within sway";
+    homepage = "https://github.com/swaywm/sway/tree/master/contrib";
+    license = licenses.mit;
+    platforms = platforms.all;
+    maintainers = with maintainers; [
+      sway-unwrapped.meta.maintainers
+      evils
+    ];
+  };
+};
+
+
+inactive-windows-transparency = python3Packages.buildPythonPackage rec {
+  # long name is long
+  lname = "inactive-windows-transparency";
+  pname = "sway-${lname}";
+  version = sway-unwrapped.version;
+
+  src = sway-unwrapped.src;
+
+  format = "other";
```

Comment:
Is this required for some build abstraction?

Reply:
you mean `format = "other";`?
this is required because buildPythonPackage defaults to looking for a `setup.py`
since we're defining an `installPhase`, `"other"` seems like the right option

Reply:
Yes this is part of the `buildPythonPackage` api. However does it run make or anything else in buildPhase? If yes, maybe also set `dontBuild = true` otherwise this looks good to me.

Reply:
> If yes

it does not do anything in build or configure phase so disabling those
same for grimshot, thanks
(can't seem to disable installCheckPhase in buildPythonPackage though)